commitment/asset.go

```diff
@@ -98,11 +101,23 @@ func parseCommon(assets ...*asset.Asset) (*AssetCommitment, error) {
 		return nil, ErrNoAssets
 	}
 
+	var assetType *asset.Type
 	maxVersion := asset.Version(0)
 	assetGenesis := assets[0].Genesis.ID()
 	assetGroupKey := assets[0].GroupKey
 	assetsMap := make(CommittedAssets, len(assets))
 	for _, asset := range assets {
+		// Set the asset type if we are handling the first asset.
+		if assetType == nil {
```

Comment:
Looking at this rule, wouldn't it make more sense to check the `TapCommitmentKey` instead? Since only assets with the same top level key can be in the same asset commitment (the type is indirectly part of the asset ID of that key or if grouped then the minting process should make sure only assets of the same type can be grouped together).

Reply:
Good point. I've added a check for consistent `TapCommitmentKey`.

I've left in the asset type check. I figured I need to pickup the asset type from one asset anyway, so I might as well check it against the other assets. We have an error type for asset type mismatch and I figured the check would be more specific and easier to debug than matching `TapCommitmentKey`. Let me know what you think!